src/layouts/Page.tsx

```diff
@@ -191,6 +187,17 @@ export const Page = (props: Props) => {
     window.location.href = window.config.SIGNUP_URL ?? import.meta.env.VITE_SIGNUP_URL
   }
 
+  useEffect(() => {
+    const hasLifespanToken = user.childs.length > 0
+
+    if (hasLifespanToken && location.pathname === routes.getLifespanToken) {
+      history.push(routes.index)
+    }
+    if (!hasLifespanToken && props.withConnection) {
```

Comment:
```
else (props.withConnection) {
} 
```

should do the same

Reply:
else {
      if (props.withConnection) {
        history.push(routes.getLifespanToken)
      }
    } 
    
    ?